### PR TITLE
feat: test that ipc verbosity is off

### PR DESCRIPTION
**Pull request by Shou**
We test that `ipc.config.silent` is set to `true` so that it doesn't
spam `stdout` for users.

Changelog-Entry: Test that IPC verbosity is off
Change-Type: patch
Depends on: https://github.com/resin-io/etcher/pull/1961

### CODE REVIEW

**Comment by jhermsmeier** [comment]
@Shou this is having lint errors (due to the required modules not being used)

**Comment by Shou** [comment]
@jhermsmeier here we go

File: tests/gui/modules/image-writer.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+'use strict'
+
+const m = require('mochainon')
+const ipc = require('node-ipc')
+// eslint-disable-next-line no-unused-vars
+const imageWriter = require('../../../lib/gui/modules/image-writer')
+
+describe('Browser: imageWriter', function () {
+  it('should have the ipc config set to silent', function () {
+    m.chai.expect(ipc.config.silent).to.be.true
```
Comment (lurch) [inline tests/gui/modules/image-writer.js]: Unlike `childWriter` (which modifies it at the global scope), `imageWriter` only modifies `ipc.config.silent` inside the `.performWrite` function?  Which, unless I'm missing something, means that this test isn't checking what we expect it to?

**Comment by lurch** [comment]
(and could you rebase this on `master`?)

File: tests/gui/modules/image-writer.js (hunk 1 above)
Comment (Shou) [reply tests/gui/modules/image-writer.js]: Good catch, I've fixed this now.

File: tests/gui/modules/image-writer.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict'
+
+const m = require('mochainon')
+const ipc = require('node-ipc')
+// eslint-disable-next-line no-unused-vars
```
Comment (lurch) [inline tests/gui/modules/image-writer.js]: I guess this eslint pragma can be removed now :)

Comment (Shou) [reply tests/gui/modules/image-writer.js]: Done

**Comment by lurch** [comment]
I've just noticed that we already have a `tests/gui/modules/image-writer.spec.js` - why does this PR add a separate `tests/gui/modules/image-writer.js` ?

**Comment by Shou** [comment]
Can't believe I missed that, thanks – also infixed `.spec` to the `child-writer` test filename. Maybe I should also add a test that the filenames in `tests/` all match the pattern in a separate PR...

File: tests/gui/modules/child-writer.spec.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict'
+
+const m = require('mochainon')
+const ipc = require('node-ipc')
+// eslint-disable-next-line no-unused-vars
+const childWriter = require('../../../lib/gui/modules/child-writer')
```
Comment (jviotti) [inline tests/gui/modules/child-writer.spec.js]: Why are we requiring something we don't use? I think you can keep the linter happy by doing:

```js
require('../../../lib/gui/modules/child-writer')
```

Notice that by running the child writer, then we're running code that tries to connect to an IPC server that doesn't exist. Are no errors throws?

Comment (Shou) [reply tests/gui/modules/child-writer.spec.js]: Yup, forgot about that – fixed now! Yeah, there are no errors. `IPC_SERVER_ID` is `undefined` in `ipc.connectTo(IPC_SERVER_ID, () => { ...` which just exits silently as long as the verbosity is set to silent <sup>[1] [2]</sup>.

[1]: https://github.com/RIAEvangelist/node-ipc/blob/master/services/IPC.js#L226
[2]: https://github.com/RIAEvangelist/node-ipc/blob/master/services/IPC.js#L64

**Pull request merged**